objects/src/notes/inputs.rs

```diff
@@ -61,6 +64,28 @@ impl NoteInputs {
     }
 }
 
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for NoteInputs {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.inputs.write_into(target);
+        self.hash.write_into(target);
```

Comment:
I wonder if we should serialize just the inputs here and compute hash using `OnceCell` approach.

Reply:
https://github.com/0xPolygonMiden/miden-base/issues/345